solidity/contracts/bridge/Bridge.sol

```diff
@@ -0,0 +1,154 @@
+    ///         keccak256(fundingTxHash | fundingOutputIndex | depositorAddress).
+    ///         This mapping may contain valid and invalid deposits and the
+    ///         wallet is responsible for validating them before attempting to
+    ///         execute a sweep.
+    mapping(uint256 => DepositInfo) public unswept;
+
+    event DepositRevealed(
+        uint256 depositId,
+        bytes32 fundingTxHash,
+        uint8 fundingOutputIndex,
+        address depositor,
+        uint64 blindingFactor,
+        bytes refundPubKey,
+        uint64 amount,
+        address vault
+    );
+
+    /// @notice Used by the depositor to reveal information about their P2SH
+    ///         Bitcoin deposit to the Bridge on Ethereum chain. The off-chain
+    ///         wallet listens for revealed deposit events and may decide to
+    ///         include the revealed deposit in the next executed sweep.
+    ///         Information about the Bitcoin deposit can be revealed before or
+    ///         after the Bitcoin transaction with P2SH deposit is mined on the
+    ///         Bitcoin chain.
+    /// @param fundingTxHash The BTC transaction hash containing BTC P2SH
+    ///        deposit funding transaction
+    /// @param fundingOutputIndex The index of the transaction output in the
+    ///        funding TX with P2SH deposit, max 256
+    /// @param blindingFactor The blinding factor used in the BTC P2SH deposit,
+    ///        max 2^64
+    /// @param refundPubKey The refund pub key used in the BTC P2SH deposit
+    /// @param amount The amount locked in the BTC P2SH deposit
+    /// @param vault Bank vault to which the swept deposit should be routed
+    /// @dev Requirements:
+    ///      - `msg.sender` must be the Ethereum address used in the P2SH BTC deposit,
+    ///      - `blindingFactor` must be the blinding factor used in the P2SH BTC deposit,
+    ///      - `refundPubKey` must be the refund pub key used in the P2SH BTC deposit,
+    ///      - `amount` must be the same as locked in the P2SH BTC deposit,
+    ///      - BTC deposit for the given `fundingTxHash`, `fundingOutputIndex`
+    ///        can be revealed by `msg.sender` only one time.
+    ///
+    ///      If any of these requirements is not met, the wallet _must_ refuse
+    ///      to sweep the deposit and the depositor has to wait until the
+    ///      deposit script unlocks to receive their BTC back.
+    function revealDeposit(
+        bytes32 fundingTxHash,
+        uint8 fundingOutputIndex,
+        uint64 blindingFactor,
+        bytes calldata refundPubKey,
+        uint64 amount,
+        address vault
+    ) external {
+        uint256 depositId =
+            uint256(
+                keccak256(
+                    abi.encode(fundingTxHash, fundingOutputIndex, msg.sender)
+                )
+            );
+
+        DepositInfo storage deposit = unswept[depositId];
+        require(deposit.revealedAt == 0, "Deposit already revealed");
```

Comment:
I think it's worth describing how do we plan to manage the `unswept` mapping once the sweep is done. If we plan to remove swept deposits from here, we must be aware someone can add an already swept deposit data via `revealDeposit`. Assuming wallets are honest, this shouldn't be a problem. On the other hand, if we plan to keep values in the `unswept` mapping forever, we need to make sure `sweep` function can be called only once with given parameters to avoid adding balance twice. I think cleaning `unswept` is the better option here.

Reply:
I think we should delete them or mark them as swept. Unsure how this should look from the `swept` function perspective - the gas cost is the key.

Reply:
Please see https://github.com/keep-network/tbtc-v2/pull/82/commits/9f2063fe88941ac4249b957a94c8b0f1c7683bce.

Reply:
Had to rebase because of linter complaints, (https://github.com/keep-network/tbtc-v2/pull/82/commits/3dbbaae81226e88047ee2f5b8febe1e6d30d28f5) is the new one.